resources.c

```diff
@@ -289,12 +302,15 @@ int res_create_multi_display_surface(const char* name, int* frames, gr_surface**
         result = -8;
         goto exit;
     }
+
+    (*surface)->_png_fd = fd;
```

Comment:
Is this line intended to be here? It looks like the first item in surface array gets _png_fd assigned and then few lines lower it is assigned to NULL without ever touching the value of _png_fd. Should this be just after the loop?

Reply:
I'm not following the need to cache the file object in the surface either, the file's not going to be read from after res_create_multi_display_surface or similar functions return right? So closing in exit: irrespective of the result should be enough?

Reply:
Likewise... Initially free'ed too early and some how ended up writing like it was. What a stupid call that was :)

Fixed.